src/software/ai/hl/stp/stp.cpp

```diff
@@ -141,17 +147,56 @@ std::vector<std::shared_ptr<Tactic>> STP::assignRobotsToTactics(
     // https://github.com/saebyn/munkres-cpp is the implementation of the Hungarian
     // algorithm that we use here
 
-    if (world.friendlyTeam().numRobots() < tactics.size())
+    auto friendly_team         = world.friendlyTeam();
```

Comment:
WIth the additional of goalie handling, this function has gotten very long. Possibly out of scope for this PR and more approriate for an stp cleanup, but this function should probably be broken up in to `assignGoalie` and `assignOtherRobots`. Your call if you want to address that now or now.

Reply:
I did quickly try this, but it's actually trickier because we do some removal of tactics/robots in the goalie section that are needed for the rest.... if you have some suggestions here I'm happy to make changes here!

Reply:
Made #1109 so I can take a separate look after.